thorlcr/graph/thgraph.cpp

```diff
@@ -2674,12 +2674,9 @@ void CJobBase::startJob()
 {
     LOG(MCdebugProgress, thorJob, "New Graph started : %s", graphName.get());
     ClearTempDirs();
-    unsigned pinterval = globals->getPropInt("@system_monitor_interval",1000*60);
-    if (pinterval)
-    {
-        perfmonhook.setown(createThorMemStatsPerfMonHook(*this, getOptInt(THOROPT_MAX_KERNLOG, 3)));
-        startPerformanceMonitor(pinterval,PerfMonStandard,perfmonhook);
-    }
+    setPerformanceMonitorHook(nullptr);
```

Comment:
Its really just for safety.  I doubt startJob() would ever be called again before endJob(), but if so - I didn't want a perf thread using the hook while in the next line it could be changed.  So I first set it to nullptr (with a lock held) and then we are free to change perfmonhook to something new.